Makefile

```diff
@@ -36,12 +36,25 @@ lint: lint.sub
 push: push.sub
 test: test.sub
 
+docker-all: .image
+	docker run -ti \
+		-v $(PWD):/mine/src/github.com/kubernetes-incubator/service-catalog \
+		buildimage make .dinit all
+
+.dinit: glide.yaml
+	make init
+	echo > .dinit
+
+.image: Dockerfile
```

Comment:
I'm not sure we really need to worry about such advanced cases. There's just one image and the odds of someone running `make` twice in the same dir at the same time seems really really low.

I will however give it a more unique name.